src/app/common/grade-icon/grade-icon.component.ts

```diff
@@ -29,7 +27,10 @@ export class GradeIconComponent implements OnInit, OnChanges {
   }
 
   private updateGrade(): void {
-    this.gradeText = this.gradeService.grades[this.grade] || 'Grade';
+    this.gradeText =
+      typeof this.grade === 'string'
+        ? this.gradeService.stringToGrade(this.grade)
```

Comment:
review how they did it in the coffee file again.

stringToGrade is used to get a number, if the input is string, which is then used to access the actual gradeText from gradeService.grades

Reply:
fixed!